awx/ui_next/src/components/ListHeader/ListHeader.jsx

```diff
@@ -38,60 +37,59 @@ class ListHeader extends React.Component {
 
   handleSearch(key, value) {
     const { location, qsConfig } = this.props;
-    let params = parseQueryString(qsConfig, location.search);
-    params = mergeParams(params, { [key]: value });
-    params = replaceParams(params, { page: 1 });
-    this.pushHistoryState(params);
+    const params = parseQueryString(qsConfig, location.search);
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      ...mergeParams(params, { [key]: value }),
+      page: 1,
+    });
+    this.pushHistoryState(qs);
   }
 
   handleReplaceSearch(key, value) {
     const { location, qsConfig } = this.props;
-    const oldParams = parseQueryString(qsConfig, location.search);
-    this.pushHistoryState(replaceParams(oldParams, { [key]: value }));
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      [key]: value,
+    });
+    this.pushHistoryState(qs);
   }
 
   handleRemove(key, value) {
     const { location, qsConfig } = this.props;
-    let oldParams = parseQueryString(qsConfig, location.search);
-    if (parseInt(value, 10)) {
-      oldParams = removeParams(qsConfig, oldParams, {
-        [key]: parseInt(value, 10),
-      });
-    }
-    this.pushHistoryState(removeParams(qsConfig, oldParams, { [key]: value }));
+    const oldParams = parseQueryString(qsConfig, location.search);
+    const updatedParams = removeParams(qsConfig, oldParams, {
+      [key]: value,
+    });
+    const qs = replaceNamespacedParams(
+      qsConfig,
+      location.search,
+      updatedParams
+    );
+    this.pushHistoryState(qs);
   }
 
   handleRemoveAll() {
-    // remove everything in oldParams except for page_size and order_by
```

Comment:
so far as I can tell, this function no longer maintains whatever page_size and order_by when remove all is clicked.  this means that clicking remove all not only removes all the filter tags, but also resets the sort and page size dropdown (which it shouldn't do).